ui/components/app/assets/asset-list/native-token/use-native-token-balance.ts

```diff
@@ -0,0 +1,85 @@
+import { useSelector } from 'react-redux';
+import {
+  showPrimaryCurrency,
+  showSecondaryCurrency,
+} from '../../../../../../shared/modules/currency-display.utils';
+import {
+  getMultichainCurrencyImage,
+  getMultichainCurrentNetwork,
+  getMultichainSelectedAccountCachedBalance,
+  getMultichainShouldShowFiat,
+} from '../../../../../selectors/multichain';
+import { getPreferences } from '../../../../../selectors';
+import { useIsOriginalNativeTokenSymbol } from '../../../../../hooks/useIsOriginalNativeTokenSymbol';
+import { PRIMARY, SECONDARY } from '../../../../../helpers/constants/common';
+import { useUserPreferencedCurrency } from '../../../../../hooks/useUserPreferencedCurrency';
+import { useCurrencyDisplay } from '../../../../../hooks/useCurrencyDisplay';
+import { TokenWithBalance } from '../asset-list';
+
+export const useNativeTokenBalance = () => {
+  const showFiat = useSelector(getMultichainShouldShowFiat);
+  const primaryTokenImage = useSelector(getMultichainCurrencyImage);
+  const { useNativeCurrencyAsPrimaryCurrency } = useSelector(getPreferences);
+  const { chainId, ticker, type, rpcUrl } = useSelector(
```

Comment:
Just mentioning that`useNativeCurrencyAsPrimaryCurrency` will be removed in the aggregated balance feature;

Reply:
Yup, this is definitely something that we are going to have conflicts with, hoping it should just be an easy swap. Will keep this comment open to keep eyes on this